app/entry.server.tsx

```diff
@@ -9,9 +10,38 @@ export default async function handleRequest(
   responseHeaders: Headers,
   remixContext: EntryContext,
 ) {
+  const {nonce, header, NonceProvider} = createContentSecurityPolicy({
+    defaultSrc: [
+      "'self'",
+      'localhost:*',
+      'https://cdn.shopify.com',
+      'https://d3hw6dc1ow8pp2.cloudfront.net',
+      'https://d3g5hqndtiniji.cloudfront.net',
+      'data:',
+    ],
+    styleSrc: [
+      "'self'",
+      "'unsafe-inline'",
+      'https://cdn.shopify.com',
+      'localhost:*',
+      'https://d3hw6dc1ow8pp2.cloudfront.net',
```

Comment:
Interesting that the merchant will have to need to know to add these.

Reply:
Indeed, but I can't think of another way to do it :pensive: